libr/bin/p/bin_qnx.c

```diff
@@ -0,0 +1,282 @@
+	sdb_set (db, "qnx.code_index", sdb_fmt ("0x%x", lmfh->code_index), 0);
+	sdb_set (db, "qnx.stack_index", sdb_fmt ("0x%x", lmfh->stack_index), 0);
+	sdb_set (db, "qnx.heap_index", sdb_fmt ("0x%x", lmfh->heap_index), 0);
+	sdb_set (db, "qnx.argv_index", sdb_fmt ("0x%x", lmfh->argv_index), 0);
+	sdb_set (db, "qnx.code_offset", sdb_fmt ("0x%x", lmfh->code_offset), 0);
+	sdb_set (db, "qnx.stack_nbytes", sdb_fmt ("0x%x", lmfh->stack_nbytes), 0);
+	sdb_set (db, "qnx.heap_nbytes", sdb_fmt ("0x%x", lmfh->heap_nbytes), 0);
+	sdb_set (db, "qnx.image_base", sdb_fmt ("0x%x", lmfh->image_base), 0);
+	return true;
+}
+
+/*
+ * Verifies the magic of the binary file
+ * @param buffer and length of signature
+ * @return bool outcome of the verification
+ */ 
+static bool check_bytes(const ut8 *buf, ut64 length) {
+	if (!buf || length < 6) {
+		return false;
+	}
+	return (!memcmp (buf, QNX_MAGIC, 6));
+}
+
+// Frees the bin_obj of the binary file
+static int destroy(RBinFile *bf) {
+	QnxObj *qo = bf->o->bin_obj;
+	r_list_free (qo->sections);
+	bf->o->bin_obj = NULL;
+	free (qo);
+	return true;
+}
+
+/*
+ * Method that loads the binary file into bin_obj
+ */  
+static bool load_bytes(RBinFile *bf, void **bin_obj, const ut8 *buf, ut64 sz, ut64 loadaddr, Sdb *sdb){
+	lmf_record *lrec = R_NEW0 (lmf_record);
+	lmf_resource *lres = R_NEW0 (lmf_resource);
+	lmf_data *ldata = R_NEW0 (lmf_data);
+	ut64 offset = QNX_RECORD_SIZE;
+	RList *ret = NULL;
+	QnxObj *qo = R_NEW0 (QnxObj);
+	if (!qo) {
+		return false;
+	}
+	if (!(ret = r_list_new ())) {
+		return false;
+	}
+	qo->kv = sdb_new0 ();
+	if (!qo->kv) {
+		free (qo);
+		return false;
+	}
+	// Read the first record
+	r_buf_fread_at (bf->buf, 0, (ut8 *) lrec, "iiii", 1);
+	// Load the header
+	lmf_header_load (&qo->lmfh, bf->buf, qo->kv);
+	offset += lrec->data_nbytes;
+	
+	for( ;; ) {
+		(void) r_buf_fread_at (bf->buf, offset, (ut8 *) lrec, "iiii", 1);
+		offset += sizeof( lmf_record );
+
+        if( lrec->rec_type == LMF_IMAGE_END_REC ) {
+            break;
+        }
+		else if (lrec->rec_type == LMF_RESOURCE_REC) {
+			RBinSection *ptr = NULL;
+			r_buf_fread_at (bf->buf, offset, (ut8 *) lres, "iccc", 1);
+			if (!(ptr = R_NEW0 (RBinSection))) {
+				return false;
+			}
+			ptr->name = strdup ("LMF_RESOURCE");
+			ptr->paddr = offset;
+			ptr->vsize = lrec->data_nbytes - sizeof (lmf_resource);
+			ptr->size = ptr->vsize;
+			ptr->add = true;
+		 	r_list_append (ret, ptr);
+		}
+		else if (lrec->rec_type == LMF_LOAD_REC) {
+			RBinSection *ptr = NULL;
+			r_buf_fread_at (bf->buf, offset, (ut8 *) ldata, "ii", 1);
+			if (!(ptr = R_NEW0 (RBinSection))) {
+				return false;
+			}
+			ptr->name = strdup ("LMF_LOAD");
+			ptr->paddr = offset;
+			ptr->vaddr = ldata->offset;
+			ptr->vsize = lrec->data_nbytes - sizeof (lmf_data);
+			ptr->size = ptr->vsize;
+			ptr->add = true;
+		 	r_list_append (ret, ptr);
+		}
+        else if (lrec->rec_type == LMF_RW_END_REC) {
```

Comment:
bad indentation , the } must be in the same line as the else

Reply:
done